container-images/scripts/build_llama_and_whisper.sh

```diff
@@ -115,7 +124,8 @@ main() {
   dnf clean all
   rm -rf /var/cache/*dnf* /opt/rocm-*/lib/llvm \
     /opt/rocm-*/lib/rocblas/library/*gfx9* \
-    /opt/rocm-*/lib/hipblaslt/library/*gfx9* llama.cpp whisper.cpp
+    /opt/rocm-*/lib/hipblaslt/library/*gfx9* llama.cpp whisper.cpp \
+    /usr/lib64/rocm/gfx9* 
```

Comment:
This might make it small enough :)

It's actually a restriction on the storage space available on our github runner, but it's a welcome restruction, we don't want 20GB container images

One could always contribute a rocm-legacy-fedora type image for gfx9 if they had the motivation to do so.

Reply:
We should actually simplify the above line to:

/opt/rocm-*/lib/*/library/*gfx9*

not necessary in this PR or anything.

Reply:
we'll find out if it's small enough soon! :smile:

Reply:
@ericcurtin it appears that the build is still angry about it :thinking:

Reply:
Yeah I guess you just gotta podman run into it and look around and see what's big, gfx9 was enough to remove for the UBI versions. Trim, test to make sure you didn't break the ROCm support, then add the "rm -fr" here.

Reply:
I'm certainly glad GitHub complains about it, don't want humongous images, keeps us on our toes 😄

Reply:
@ericcurtin question - what's the size goal I'm trying to hit?

Reply:
I can't remember exactly I think it's 10G which is pretty damn big. The ROCm images are the only images that suffer from this, I've never had to trim a CUDA one for example.

Reply:
It's arguably worth logging an issue in ROCm upstream, I find it hard to believe all those gfx9 files are completely unique, I'm sure there's loads of duplicate data in those files that could be shared using some common files or some other form of de-duplication.

Reply:
It leaves a bad first impression for ramalama if users have to pull huge container images just to get going, some people won't wait around for hours for a large image to be pulled.

Reply:
We could also do container images like rocm-gfx9-fedora, rocm-gfx10-fedora, etc. We may be able to detect what generation we are pre-pull